matplotlib2tikz/axes.py

```diff
@@ -389,7 +389,10 @@ def _get_ticks(data, xy, ticks, ticklabels):
         pgfplots_ticks.append(tick)
         # store the label anyway
         label = ticklabel.get_text()
-        pgfplots_ticklabels.append(label)
+        if ticklabel.get_visible():
+            pgfplots_ticklabels.append(label)
+        else:
```

Comment:
Aha, I see. Well, a bit of this text should go as documentation in the code, otherwise I see myself remove this bit in half a year, having forgotten what it was all about. Perhaps something along the lines of `By default, Pgfplots plots ticks, so no explicit labels aren't required. If _no_ ticks are forced, we need "xticklabels={}".`